src/helpers/prss.rs

```diff
@@ -94,14 +94,68 @@ impl PrssSpace {
         let (l, r): (F, F) = self.generate_fields(index);
         l + r
     }
+
+    /// Turn this space into a pair of random number generators, one that is shared
+    /// with the left and one that is shared with the right.
+    /// Nothing prevents this from being used after calls to generate values,
+    /// but it should not be used in both ways.
+    #[must_use]
+    pub fn as_rngs(&self) -> (PrssRng, PrssRng) {
+        (
+            PrssRng {
+                generator: self.left.clone(),
+                counter: 0,
+            },
+            PrssRng {
+                generator: self.right.clone(),
+                counter: 0,
+            },
+        )
+    }
+}
+
+/// An implementation of `RngCore` that uses the same underlying `Generator`.
+/// For use in place of `PrssSpace` where indexing cannot be used, such as
+/// in APIs that expect `Rng`.
+#[allow(clippy::module_name_repetitions)]
+pub struct PrssRng {
+    generator: Generator,
+    counter: u128,
 }
 
+impl RngCore for PrssRng {
+    #[allow(clippy::cast_possible_truncation)]
+    fn next_u32(&mut self) -> u32 {
+        self.next_u64() as u32
+    }
+
+    // This implementation wastes half of the bits that are generated.
```

Comment:
I guess we could keep `Option<u64>` in `PrssRng`, store it as 64 MSB from `v` left after truncation and yield it when `next_u64` is called if we really need to

Reply:
Yeah, as noted, I don't think that this is a real concern, absent performance metrics that suggest otherwise.